00XX-common-ledger-protocol/00XX-common-ledger-protocol.md

```diff
@@ -0,0 +1,273 @@
+
+ILP payment packets are attached to the side protocol data under the protocol
+`ilp`. 
+
+#### Response
+
+- `Response` is returned if the peer acknowledges the `Prepare`. This means the
+  transfer is now `prepared` and has been applied to the balance. It may carry
+side protocol data, for example a payment channel claim.
+
+- `Error` is returned if the peer does not accept the transfer. This could be
+  because it is incompatible with an existing transfer with the same ID (see
+Idempotency below), or because it exceeds some constraint your peer has placed.
+The amount could be too high, or could exceeds the balance.
+
+#### Idempotency
+
+If a transfer with the given ID already exists in any state, a new transfer
+should not be created. If the packet (including side protocol data) matches an
+existing transfer exactly and that transfer is in the `prepared` state, a
+`Response` should be returned. If the packet (including side protocol data) shares
+the ID of an existing transfer but other details do not match, an `Error`
+should be returned. If the packet (including side protocol data) exactly
+matches an existing transfer but that transfer is in the `fulfilled` state or
+the `rejected` state, an `Error` should be returned.
+
+#### Expiry
+
+After the `expiresAt` date is reached, the transfer can no longer be fulfilled.
+If one party on the CLP connection is keeping authoritative state, they should
+send a `Reject` request to the other party. If both parties are keeping
+authoritative state, they can independently expire the transfer (set the state
+to `rejected` and roll it back) automatically.
+
+### Fulfill
+
+`Fulfill` is used to change an existing transfer from the `prepared` state to
+the `fulfilled` state. The packet-specific data of `Fulfill` is made up of `transferId`
+and `fulfillment`. `Fulfill` is a request with side effects: it changes the
+state of the transfer and finalizes the transfer's balance update. If the
+transfer is already fulfilled, no changes are applied.
+
+`transferId` is a 128-bit unique ID, which references an existing transfer.
+`fulfillment` is a 256-bit integer, containing the preimage of the
+`executionCondition` of the referenced transfer.
+
+#### Response
+
+A `Fulfill` request must be sent from the receiver of the referenced transfer,
+not the sender.
+
+A `Fulfill` request is successful if `transferId` references an existing
+transfer in the `prepared` state, the `fulfillment` is the SHA-256 preimage of
+the referenced transfer's `executionCondition`, and the current time is before
+the referenced transfer's `expiresAt`.
+
+A `Fulfill` request is also successful if the transfer is in the `fulfilled`
+state, and this packet matches the one that fulfilled the transfer,
+including side protocols but not including requestId.
+
+- `Response` is returned if the request is successful.  data to be passed back.
+  This means the transfer's state is now `fulfilled`, and its balance change
+has been finalized.
```

Comment:
That depends on who is authoritative; if the slave fulfills, then this statement is correct, but if the master fulfills, the balance change is already finalized, even if the slave doesn't respond, or responds with an error. Note that this is independent of which party is the WebSocket server and which one is the WebSocket client.